platform/src/main/java/org/stellar/anchor/platform/payment/observer/stellar/StellarPaymentObserver.java

```diff
@@ -59,14 +66,41 @@ public void shutdown() {
     this.stream.close();
   }
 
-  public SSEStream<OperationResponse> watch() {
-    PaymentsRequestBuilder paymentsRequest =
-        server.payments().includeTransactions(true).order(RequestBuilder.Order.ASC);
+  String fetchStreamingCursor() {
+    // Use database value, if any.
     String lastToken = paymentStreamerCursorStore.load();
     if (lastToken != null) {
-      paymentsRequest.cursor(lastToken);
+      return lastToken;
     }
 
+    // Otherwise, fetch the latest value from the network.
+    Page<OperationResponse> pageOpResponse;
+    try {
+      pageOpResponse =
+          server.payments().order(RequestBuilder.Order.DESC).limit(MIN_RESULTS).execute();
+    } catch (IOException e) {
+      Log.errorEx("Error fetching the latest /payments result.", e);
+      return null;
+    }
+
+    if (pageOpResponse == null
+        || pageOpResponse.getRecords() == null
+        || pageOpResponse.getRecords().size() == 0) {
+      return null;
+    }
+    return pageOpResponse.getRecords().get(0).getPagingToken();
+  }
+
+  public SSEStream<OperationResponse> watch() {
+    String latestCursor = fetchStreamingCursor();
+    PaymentsRequestBuilder paymentsRequest =
+        server
+            .payments()
+            .includeTransactions(true)
+            .cursor(latestCursor)
+            .order(RequestBuilder.Order.ASC)
+            .limit(MAX_RESULTS);
```

Comment:
Whenever fetching results from the network, we can manually set the limit to (up to) 200 results. This helps to make the network calls more efficient.

Cc @lijamie98, @erika-sdf, @stfung77.